fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -77,6 +79,21 @@
             SYSTEM_CONFIG.getPackageVariableName("lucene_search_timeout_ms"), 600000
     );
 
+    public static final LRUQueryCache LUCENE_QUERY_CACHE;
+    static {
+        // Lucene 5.3.0 disables query caching.  Create cache for use here.
+        // min of 32MB or 5% of the heap size.  Same default as Lucene 5.4.0.
+        final long maxRamBytesUsed = SYSTEM_CONFIG.getLongProperty(
+            SYSTEM_CONFIG.getPackageVariableName("lucene_cache_size_bytes"),
+            Math.min(1L << 25, Runtime.getRuntime().maxMemory() / 20));
```

Comment:
Ultimately, checkstyle and PR reviews are the arbiters of truth, and the IDE configs are really only a guide to help people get up and running with a sane starting point. If there are some quick wins we can get to handle unambiguous common cases, I'm all for automating wherever we can to make everyone's life easier. Unfortunately, not everything has a clear "doing it this way makes it easier to read and understand", which is the ultimate goal.

I'm not familiar with the Maven plugins that do auto-formatting, but if someone wants to take a stab at hooking one up and starting with some simple rules (eg. indent levels and spaces instead of tabs) to keep changes generally in line with the current codebase, I would be happy to pull that in.